docs/howto/manage-storage.md

```diff
@@ -164,7 +96,107 @@ The storage will not be available immediately after that call - the charm should
 observe the `storage-attached` event and handle any remaining setup once Juju
 has attached the new storage.
 
-### Write unit tests
+## Manage storage for a Kubernetes charm
+
+### Define the storage
+
+Define a `storage` section in `charmcraft.yaml` that lists the storage volumes and information about each storage. Also specify where to mount the storage in the workload container.
+
+For example, for a transient filesystem storage that is at least 1GB in size:
+
+```yaml
+storage:
+  cache:
+    type: filesystem
+    description: Somewhere to cache files locally.
+    minimum-size: 1G
+    properties:
+      - transient
+
+containers:
+  web-service:
+    resource: app-image
+    mounts:
+      - storage: cache
+        location: /var/cache
+```
+
+Juju mounts the storage at `/var/cache` in the workload container.
+
+### Access the storage
+
+In your charm's `__init__` method, observe the [storage-attached](ops.StorageAttachedEvent) event:
+
+```python
+framework.observe(self.on["cache"].storage_attached, self._update_configuration)
+```
+
+In this example, we use a holistic event handler called `_update_configuration`. Alternatively, you could use a dedicated handler for the storage-attached event. To learn more about the different approaches, see [](#holistic-vs-delta-charms).
+
+Next, access the storage in `_update_configuration`. Don't hard-code the location of the storage. Instead, get the location from [`ContainerMeta.mounts`](ops.ContainerMeta.mounts) for the workload container:
```

Comment:
I think we should emphasise that the location shouldn't be hard-coded. I don't think it needs to be an admonition box, but perhaps:

```suggestion
Next, access the storage in `_update_configuration`. **Don't hard-code the location of the storage.** Instead, get the location from [`ContainerMeta.mounts`](ops.ContainerMeta.mounts) for the workload container:
```